packages/office-addin-usage-data/src/usageData.ts

```diff
@@ -271,11 +271,12 @@ export class OfficeAddinUsageData {
    */
   public maskFilePaths(err: Error): Error {
     try {
-      const regexRemoveUserFilePaths = /\/(.*)\//gmi;
-      const regexRemoveUserFilePathsFromStack = /\w:\\(?:[^\\\s]+\\)+/gmi;
-      err.message = err.message.replace(regexRemoveUserFilePaths, "");
-      err.stack = err.stack.replace(regexRemoveUserFilePaths, "");
-      err.stack = err.stack.replace(regexRemoveUserFilePathsFromStack, "");
+      const regexRemoveUserFilePaths = /[\/\\](.*)[\/\\]/gmi;
+      const regexRemoveAbsoluteUserFilePathsFromStack = /\w:\\(?:[^\\\s]+\\)+/gmi;
+      const regexRemoveFirstFilePathFromStack = /\\(.*)\./i;
```

Comment:
I'm not clear on what the second two expressions are for.

Reply:
Wouldn't it be better to mask the user folder name, e.g. a path like "c:\users\adamk\..." would be turned into "c:\users\****\..."

I personally cannot read these regexes to know easily what they are doing. Can we please add unit tests for these so we can have tests which show what they do and that it is the desired behavior?

Reply:
The first expression is for using on the error message.

The second and third are for using in the stack of the error. The second is to remove absolute paths from the stack and only keep the internal paths, while the third is to trim the first file path to only have the extension in it (because of UID problems in the filename).

Reply:
@akrantz yes, regex are pretty hard to read, but there are unit tests for they already and I added some more inside this PR.
By the users you mean to mask by users and not just by C:?

Reply:
What does the "user" have to do with any of these . . . seems like the operate on any file path?
Why specifically absolute files paths?
Why just the "first file path" . . . seems like it would affect any file path?

Reply:
The first file path only applies to the first one because the is no `g` flag. You can that the other 2 regex have it, so there are applied to all matchs.

The user is not applied right now, I just check if the path is in the format `LETTER:` like C:

Absolute file path: C:\path_to_file
Internal path: /src/index.js

Reply:
Why do we need a separate expression for the first case vs the others?

Reply:
@millerds for the first case I'm only leaving the file extension, to avoid having the UID.
For all the other on the stack I'm leaving the filename, as it is needed to debug (just the extension would not be enough).
This should fix most of the problems, as I think the UID is the most common on the first file of the stack error